src/main/java/ch/njol/skript/expressions/ExprAdvancementMessage.java

```diff
@@ -0,0 +1,131 @@
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import io.papermc.paper.advancement.AdvancementDisplay;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.event.HoverEvent;
+import net.kyori.adventure.text.format.TextColor;
+import org.bukkit.Bukkit;
+import org.bukkit.advancement.Advancement;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerAdvancementDoneEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Advancement Message")
+@Description("The message of an advancement in the <a href=\\\"./events.html#advancement_complete\\\">advancement complete event</a>.")
+@Examples("set advancement message to \"%player% completed an advancement!\"")
+@Since("INSERT VERSION")
+public class ExprAdvancementMessage extends SimpleExpression<String> {
+
+	static {
+		if (Skript.classExists("net.kyori.adventure.text.Component") && Skript.classExists("io.papermc.paper.advancement.AdvancementDisplay"))
+			Skript.registerExpression(ExprAdvancementMessage.class, String.class, ExpressionType.SIMPLE, "[the] advancement message");
+	}
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!getParser().isCurrentEvent(PlayerAdvancementDoneEvent.class))
+			Skript.error("The advancement message expression can only be used inside of the advancement complete event.");
+		return getParser().isCurrentEvent(PlayerAdvancementDoneEvent.class);
+	}
+
+	@Override
+	@Nullable
+	protected String[] get(Event event) {
+		if (event instanceof PlayerAdvancementDoneEvent) {
+			PlayerAdvancementDoneEvent advEvent = (PlayerAdvancementDoneEvent) event;
+			if (advEvent.message() != null)
+				return new String[]{Bukkit.getUnsafe().legacyComponentSerializer().serialize(advEvent.message())};
+		}
+		return null;
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		switch (mode) {
+			case SET:
+			case DELETE:
+			case RESET:
+				return CollectionUtils.array(String.class);
+			default:
+				return null;
+		}
+	}
+
+	@Override
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		if (event instanceof PlayerAdvancementDoneEvent) {
+			PlayerAdvancementDoneEvent advEvent = (PlayerAdvancementDoneEvent) event;
+			switch (mode) {
+				case SET:
+					if (delta != null)
+						advEvent.message(Component.text((String) delta[0]));
+					break;
+				case DELETE:
+					if (delta != null)
+						advEvent.message(Component.text(""));
```

Comment:
Does using an empty component work here? I think it's like `Component.empty()` or something